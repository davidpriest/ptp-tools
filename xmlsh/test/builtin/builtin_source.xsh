# Test of source (.) command

if [ -z "$_SOURCE" ] ; then
	echo top level invocation of $(xfile -b $0)
	_SOURCE=source
	. $0 
	echo _VAR is $_VAR 	
	unset _SOURCE
	unset _VAR
	
	# Try using source command instead of .
	_SOURCE=source
	source $0 
	echo _VAR is $_VAR 	
	unset _SOURCE
	unset _VAR
else
	echo sourced invocation of $(xfile -b $0)
	echo _SOURCE is $_SOURCE
	_VAR=var
fi
