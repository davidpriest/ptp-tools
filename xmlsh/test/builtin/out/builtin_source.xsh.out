top level invocation of builtin_source
sourced invocation of builtin_source
_SOURCE is source
_VAR is var
sourced invocation of builtin_source
_SOURCE is source
_VAR is var
